linen_examples/pixelcnn/main.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Main file for running the example.
+
+This file is intentionally kept short. The majority for logic is in libraries
+than can be easily tested and imported in Colab.
+"""
+
+from absl import app
+from absl import flags
+from absl import logging
+
+from clu import platform
+import sample
+import train
+import jax
+from ml_collections import config_flags
+import tensorflow as tf
+
+FLAGS = flags.FLAGS
+
+config_flags.DEFINE_config_file(
+    'config', 'configs/default.py', 'Training configuration.', lock_config=True)
+flags.DEFINE_string('workdir', None, 'Work unit directory.')
+flags.DEFINE_bool('sample', False, 'Sample from a model in workdir.')
+flags.DEFINE_string('jax_backend_target', None,
+                    'JAX backend target to use. Can be used with UPTC.')
+flags.mark_flags_as_required(['config', 'workdir'])
+
+
+def main(argv):
+  if len(argv) > 1:
+    raise app.UsageError('Too many command-line arguments.')
+
+  # Hide any GPUs form TensorFlow. Otherwise TF might reserve memory and make
+  # it unavailable to JAX.
+  tf.config.experimental.set_visible_devices([], 'GPU')
+
+  if FLAGS.jax_backend_target:
+    logging.info('Using JAX backend target %s', FLAGS.jax_backend_target)
+    jax.config.update('jax_xla_backend', 'tpu_driver')
+    jax.config.update('jax_backend_target', FLAGS.jax_backend_target)
+
+  logging.info('JAX host: %d / %d', jax.host_id(), jax.host_count())
+  logging.info('JAX local devices: %r', jax.local_devices())
+
+  # Add a note so that we can tell which task is which JAX host.
```

Comment:
I know these lines weren't added in this PR, but I find there to be a bit much boilerplate here, specifically some of the stuff that calls into CLU that isn't really configurable here for most users.

Could we extract lines 51-4 and 59-64 into a function, perhaps `setup_clu()`?

Reply:
Can we first bring all examples into a consistent state and then remove common boilerplate? 

I will remove line 51-54 for now.
Lines 59-63: For running internally I find these quite useful (as it shows up in the UI) but I'm not yet ready to make this a default for every user of `clu`.